src/index.ts

```diff
@@ -57,40 +62,53 @@ const init: InitCMP = ({
 
 	switch (framework) {
 		case 'ccpa':
-			CMP = CCPA;
+			activeCMP = CCPA;
 			break;
 		case 'aus':
-			CMP = AUS;
+			activeCMP = AUS;
 			break;
 		case 'tcfv2':
 		default:
 			// default is also 'tcfv2'
-			CMP = TCFv2;
+			activeCMP = TCFv2;
 			break;
 	}
 
 	setCurrentFramework(framework);
 
-	CMP.init(pubData ?? {});
+	activeCMP.init(pubData ?? {});
 	resolveInitialised();
 };
 
 const willShowPrivacyMessage: WillShowPrivacyMessage = () =>
-	initialised.then(() => CMP?.willShowPrivacyMessage() ?? false);
+	initialised.then(() => activeCMP?.willShowPrivacyMessage() ?? false);
+
+const willShowPrivacyMessageSync = () => {
+	if (_willShowPrivacyMessage !== void 0) {
```

Comment:
Why not just use `undefined`? 

> The `void` operator is often used merely to obtain the `undefined` primitive value, usually using "`void(0)`" (which is equivalent to "`void 0`"). In these cases, the global variable [`undefined`][] can be used.
>
> _Source: [MDN: void](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void)_

[`undefined`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined

```suggestion
	if (_willShowPrivacyMessage !== undefined) {
```

Reply:
`undefined` used to be able to be redefined. that's prob less dangerous now, it's an old habit. but `typeof void 0 === 'undefined'` is _always_ true

Reply:
as in, `'undefined'`, the string?

Reply:
To be clear `void([any valid expression])` and returns undefined. So `void 0` (or `void 2121` or `void Math.PI`) always = `undefined`.
This is also why some libraries would wrap in a ```function(undefined) {
// all the code here safe in the knowledge that undefined is actually undefined
}()``` 

But this is a history lesson. Since undefined it's no longer writeable I vote accepting Max's suggestion. Simpler. It's not an idiom that you see much anymore and no need to confuse the young'ns.

Reply:
I seem to recall @janua had a plan to write a jquery-style `undefined.js` library once, where all the utilities hung off `undefined`instead

Reply:
so much for my 'early adopter' approach @mxdvl!